webapp/publisher/snaps/views.py

```diff
@@ -806,6 +806,33 @@ def post_register_name():
     return flask.redirect(flask.url_for("account.get_account"))
 
 
+@publisher_snaps.route("/register-snap/json", methods=["POST"])
+@login_required
+def post_register_name_json():
+    snap_name = flask.request.form.get("snap-name")
+
+    if not snap_name:
+        return flask.jsonify({"success": False})
```

Comment:
Is the success key to help on the client side? Or could we just return an error message and a 400 status code?

Reply:
Old habit from build.snapcraft I suppose.
But sending list of errors in JSON with a proper status code probably would be enough to tell if there was an error or success in front-end.